### DISCUSSION TITLE
Prevent Materialisation if all partitions in upstream asset are not Materialised

### DISCUSSION THREAD

**Discussion by SEAN-7** [dagster-io/dagster#13032] [Q&A] [answered]
Hi guys, I've got an ETL process where Asset A fetches yesterdays data from an API. It's partitioned daily, and scheduled to run daily. Asset B is a transformed version of Asset A. It is not partitioned. It reads all the daily data from Asset A. 

I want to prevent Asset B from materialising if any of the partitions of Asset A are missing. Currently, I can go into Dagit, and hit the Materialise button on Asset B, and it will happily materialise it even though there are partitions missing on Asset A.

Does the Dagster API have a way of handling this?

Cheers
Sean

**Comment by SEAN-7** [comment, marked as answer]
You can check if all partitions in Asset A have been materialised in the asset definition of Asset B, and throw an Error if they haven't been materialised. The DagsterInstance (available on the context parameter) has a method called get_materialization_count_by_partition, that will tell you which partitions have been materialised (and how many times) for a given asset. For example:

```python
@asset(ins={"asset_a": AssetIn(partition_mapping=AllPartitionMapping())})
def asset_b(
    context: OpExecutionContext
):
    materialization_count_by_partition = context.instance.get_materialization_count_by_partition(asset_keys=[AssetKey('asset_a')]).get(AssetKey('asset_a')

    asset_a_partitions = context.asset_partition_keys_for_input('asset_a')

    # From here, it's simply a matter of checking if each partition has been materialised
```

**Reply by Tazoeur** [reply]
Starting from dagster 1.5.9, I don't have access to `context.instance.get_materialization_count_by_partition` (it may even be removed before this version, I haven't checked)

I now have to use `context.instance.get_materialized_partitions`, which I find more eloquent even if I had to pay a production bug for this.
